Microsoft Dynamics 365 (destination) documentation

mohan-h opened the pull request:
Initial commit

djwendo commented:
Thank you for pointing out the necessary edits to the code snippet. This update has been made through git.corp and merged.

mohan-h committed:
Microsoft Dynamics 365 (destination) documentation

Initial commit

mohan-h committed:
Minor cosmetic update

mohan-h committed:
Merge branch 'main' into msdynamics-dest

mohan-h committed:
Minor content addition

mohan-h committed:
Merge branch 'AdobeDocs:main' into msdynamics-dest

mohan-h committed:
Added limits section and minor updates

mohan-h committed:
Updates against review

vfalon commented on help/destinations/catalog/crm/microsoft-dynamics-365.md:
```diff
@@ -0,0 +1,213 @@
+
+This [!DNL Adobe Experience Platform] [destination](/help/destinations/home.md) leverages the [Contact Entity Reference API](https://docs.microsoft.com/en-us/dynamics365/customerengagement/on-premises/developer/entities/contact?view=op-9-1), which allows you to update identities within a segment into [!DNL Dynamics 365].
+
+[!DNL Dynamics 365] uses OAuth 2 with Authorization Grant as the authentication mechanism to communicate with the Contact Entity Reference API. Instructions to authenticate to your [!DNL Dynamics 365] instance are further below, in the [Authenticate to destination](#authenticate) section.
```
Let's wrap Contact Entity Reference API in DNL tags.

vfalon commented on help/destinations/catalog/crm/microsoft-dynamics-365.md:
```diff
@@ -0,0 +1,213 @@
+
+This [!DNL Adobe Experience Platform] [destination](/help/destinations/home.md) leverages the [Contact Entity Reference API](https://docs.microsoft.com/en-us/dynamics365/customerengagement/on-premises/developer/entities/contact?view=op-9-1), which allows you to update identities within a segment into [!DNL Dynamics 365].
+
+[!DNL Dynamics 365] uses OAuth 2 with Authorization Grant as the authentication mechanism to communicate with the Contact Entity Reference API. Instructions to authenticate to your [!DNL Dynamics 365] instance are further below, in the [Authenticate to destination](#authenticate) section.
```
Is Dynamics 365 an approved branding for it? If it's not used in the Microsoft docs, let's use [!DNL Microsoft Dynamics 365] instead.

mohan-h replied:
Yes it is even their [homepage](https://dynamics.microsoft.com/en-us/?cdn=disable) refers to it.

vfalon commented on help/destinations/catalog/crm/microsoft-dynamics-365.md:
```diff
@@ -0,0 +1,213 @@
+
+### Experience Platform Prerequisites {#prerequisites-in-experience-platform}
+
+Before activating data to the [!DNL Microsoft Dynamics 365] destination, you must have a [schema](https://experienceleague.adobe.com/docs/experience-platform/xdm/schema/composition.html), a [dataset](https://experienceleague.adobe.com/docs/platform-learn/tutorials/data-ingestion/create-datasets-and-ingest-data.html?lang=en), and [segments](https://experienceleague.adobe.com/docs/platform-learn/tu ...
```
Please use relative links to the AEP docs instead of public Experience League links.

vfalon commented on help/destinations/catalog/crm/microsoft-dynamics-365.md:
```diff
@@ -0,0 +1,213 @@
+
+Before activating data to the [!DNL Microsoft Dynamics 365] destination, you must have a [schema](https://experienceleague.adobe.com/docs/experience-platform/xdm/schema/composition.html), a [dataset](https://experienceleague.adobe.com/docs/platform-learn/tutorials/data-ingestion/create-datasets-and-ingest-data.html?lang=en), and [segments](https://experienceleague.adobe.com/docs/platform-learn/tu ...
+
+Refer to Adobe's documentation for [Segment Membership Details schema field group](https://experienceleague.adobe.com/docs/experience-platform/xdm/field-groups/profile/segmentation.html?lang=en) if you need guidance on segment statuses.
```
Please use relative links for AEP docs.

vfalon commented on help/destinations/catalog/crm/microsoft-dynamics-365.md:
```diff
@@ -0,0 +1,213 @@
+
+| Credential | Description | Example |
+| --- | --- | --- |
+| Username | Your [!DNL Dynamics 365] account username | - |
```
Let's include a generic username/password, like MyDynamicsUsername and MyDynamicsPassword. Maybe we can also link to the Dynamics 365 account management docs, if there are any.

vfalon commented on help/destinations/catalog/crm/microsoft-dynamics-365.md:
```diff
@@ -0,0 +1,213 @@
+
+To connect to this destination, follow the steps described in the [destination configuration tutorial](../../ui/connect-destination.md). In the configure destination workflow, fill in the fields listed in the two sections below.
+
+![Catalog](../../assets/catalog/crm/microsoft-dynamics-365/catalog.png)
```
There is no need for this screenshot, as this process is already illustrated in the destination config tutorial.

mohan-h replied:
Screenshot has been removed.

vfalon commented on help/destinations/catalog/crm/microsoft-dynamics-365.md:
```diff
@@ -0,0 +1,213 @@
+
+*  **[!UICONTROL Name]**: A name by which you will recognize this destination in the future.
+*  **[!UICONTROL Description]**: A description that will help you identify this destination in the future.
+*  **[!UICONTROL Environment URL]**: Your [!DNL Dynamics 365] Environment URL.
```
Can we add a link to the Dynamics 365 docs where users can see how to obtain the Environment URL?

vfalon replied:
Let's add here the link that you used in the "Gather Dynamics 365 credentials" section, referring to the env URL.

vfalon commented on help/destinations/catalog/crm/microsoft-dynamics-365.md:
```diff
@@ -0,0 +1,213 @@
+1. In the Mapping step, click **[!UICONTROL Add new mapping]**, you will see a new mapping row on the screen.
+![Add new mapping](../../assets/catalog/crm/microsoft-dynamics-365/add-new-mapping.png)
+
+1. In the select source field window, when selecting the source field choose the **[!UICONTROL Select identity namespace]** category and add the mappings desired.
```
Is the contactId namespace the only supported one? If so, let's call that out instead of  "add the mappings desired."

vfalon commented on help/destinations/catalog/crm/microsoft-dynamics-365.md:
```diff
@@ -0,0 +1,213 @@
+![Source mapping](../../assets/catalog/crm/microsoft-dynamics-365/source-mapping.png)
+
+
+1. In the "[!UICONTROL Select target field]" window, select the target field 
```
Line 133: In the "[!UICONTROL Select target field]" window, select the type of target field that you want to map your source field to.

vfalon commented on help/destinations/catalog/crm/microsoft-dynamics-365.md:
```diff
@@ -0,0 +1,213 @@
+        |`person.name.lastName`|`LastName`|
+        |`personalEmail.address`|`Email`|
+
+1. An example using these mappings is shown below:
```
Line 152: this line looks like it should be indented, to be in line with the example above. It's not a separate step.

mohan-h replied:
Not too sure about all of this as had originally intended to have separate level 1 points for identity namespace and custom attributes mapping nevertheless have done it.

vfalon commented on help/destinations/catalog/crm/microsoft-dynamics-365.md:
```diff
@@ -0,0 +1,213 @@
+1. An example using these mappings is shown below:
+![Target mapping](../../assets/catalog/crm/microsoft-dynamics-365/mappings.png)
+
+> [!IMPORTANT]
```
This note seems like it comes out of nowhere. I don't understand what this is supposed to mean. Let's add some context. What does a date or timestamp mapping look like? Why would someone do that? Let's explain this in a dedicated section.

mohan-h replied:
Line 141 :  Refer to [[!DNL Dynamics 365] documentation](https://docs.microsoft.com/en-us/dynamics365/customerengagement/on-premises/developer/entities/contact?view=op-9-1#entity-properties) for a comprehensive list of supported attributes.

These have a number of date / timestamp fields. The user can map to these fields but if the value is left empty then push to dynamics would fail.

Have moved the block just below the above reference link.

vfalon replied:
Sorry, this still doesn't make much sense to me. The supported identity for this destination is `contactId`. Why would anyone map anything to a date or timestamp? Where did this information come from? Was this asked by PM/Engineering?

mohan-h replied:
Yes, this has been specifically pointed out by engineering. There could be multiple use cases we might want to store the date of the last communication that was sent to them etc.

vfalon commented on help/destinations/catalog/crm/microsoft-dynamics-365.md:
```diff
@@ -0,0 +1,213 @@
+
+> [!IMPORTANT]
+>
+> The value used for the Mapping ID should exactly match the name of the custom field attribute created within [!DNL Dynamics 365].
```
Let's add a link to the Dynamics docs where they explain where to find the custom field name.

vfalon commented on help/destinations/catalog/crm/microsoft-dynamics-365.md:
```diff
@@ -0,0 +1,213 @@
+1. Monitor the segment summary and ensure that the count of profiles corresponds to the count created within the segment.
+![Segment](../../assets/catalog/crm/microsoft-dynamics-365/segment.png)
+
+1. Login to the Dynamics 365 website, then navigate to the **[!DNL Apps]** > **[!DNL Contacts]** page and check if the profiles from the segment have been added. You will also notice the segment status from Experience Platform has been updated against the corresponding custom field attribute that was provided in the **Mapping ID** field during the **[!UICONTROL Activate destination]** > **[!UICON ...
```
Line 188: _Log in_ instead of _Login_.

vfalon commented on help/destinations/catalog/crm/microsoft-dynamics-365.md:
```diff
@@ -0,0 +1,213 @@
+## Errors and troubleshooting {#errors-and-troubleshooting}
+
+### Authentication errors encountered while pushing events to destination {#auth-errors}
+During execution the [!DNL Dynamics 365] access token might become invalid, in such scenarios you will obtain the error message below, To resolve re-authenticate and reattempt the execution.
```
Line 198: "During activation, the [!DNL Dynamics 365] access token might become invalid, resulting in the following error message: `Authentication error reported while pushing events to the destination. Please check authentication credentials and try again.`

To fix this error, re-authenticate to the destination and create a new activation flow".

Let's spell out the error messages in the docs. Users might search for those error messages and they won't see them as easily in screenshots.

vfalon commented on help/destinations/catalog/crm/microsoft-dynamics-365.md:
```diff
@@ -0,0 +1,213 @@
+![Error](../../assets/catalog/crm/microsoft-dynamics-365/error-auth.png)
+
+### Unknown errors encountered while pushing events to destination {#unknown-errors}
+When checking a dataflow run, if you see the error message below, verify that the Mapping ID that you provided in [!DNL Dynamics 365] for your Platform segment is valid and exists within [!DNL Dynamics 365].
```
Please rephrase this section similar to the example I provided for the one above.

vfalon reviewed (changes requested):
Please see my comments and I'll take another look once you finish going through them. Thanks!

mohan-h committed:
Updates post review

mohan-h committed:
Content updates

mohan-h committed:
Updates post review

mohan-h committed:
Merge branch 'AdobeDocs:main' into msdynamics-dest

vfalon commented on help/destinations/catalog/crm/microsoft-dynamics-365.md:
```diff
@@ -0,0 +1,218 @@
+description: The Microsoft Dynamics 365 destination allows you to export your account data and activate it within Microsoft Dynamics 365 for your business needs.
+---
+
+# [!DNL Microsoft Dynamics 365] connection
```
Let's improve the consistency of the branding across this page. Let's use `[!DNL Microsoft Dynamics 365]` in the title and in the first occurrence from the Overview section, and use `[!DNL Dynamics 365]` in all subsequent occurences, so we don't repeat "Microsoft" randomly across the page.

mohan-h replied:
The intent was for each section to start with `[!DNL Microsoft Dynamics 365]` with subsequent occurrences within that section as `[!DNL Dynamics 365]` without the "Microsoft".

vfalon commented on help/destinations/catalog/crm/microsoft-dynamics-365.md:
```diff
@@ -0,0 +1,218 @@
+Refer to the [!DNL Dynamics 365] documentation to [create a field (attribute)](https://docs.microsoft.com/en-us/dynamics365/customerengagement/on-premises/customize/create-edit-fields?view=op-9-1) if you need additional guidance.
+
+An example setup within [!DNL Dynamics 365] is shown below:
+![Target mapping](../../assets/catalog/crm/microsoft-dynamics-365/dynamics-365-fields.png)
```
One thing that applies to all destinations. To improve readability and accessibility for all our docs, we have been asked to use clearer alt-text for images. Please review all the alt-text in the page and update it to a form similar to this:

"Dynamics 365 UI screenshot showing the Fields screen."

Use clear sentences that describe as accurate as possible what is shown in the image. This is especially useful for users who are unable to see the image and/or use screen readers.

vfalon commented on help/destinations/catalog/crm/microsoft-dynamics-365.md:
```diff
@@ -0,0 +1,218 @@
+
+|Target Identity|Example|Description|Considerations|
+|---|---|---|---|
+| `contactId` | `7eb682f1-ca75-e511-80d4-00155d2a68d1` | Unique identifier for a contact.| **Mandatory**, Refer to the [[!DNL Microsoft Dynamics 365] documentation](https://docs.microsoft.com/en-us/dynamics365/customerengagement/on-premises/developer/entities/contact?view=op-9-1) for further details. |
```
No need to wrap the example in code.

vfalon commented on help/destinations/catalog/crm/microsoft-dynamics-365.md:
```diff
@@ -0,0 +1,218 @@
+1. In the "[!UICONTROL Select source field]" window, when selecting the source field choose the **[!UICONTROL Select identity namespace]** category and select `contactId`.
+![Source mapping](../../assets/catalog/crm/microsoft-dynamics-365/source-mapping.png)
+
+1. In the "[!UICONTROL Select target field]" window, select the type of target field that you want to map your source field to.
```
No need for quotes on the UICONTROLs.

vfalon commented on help/destinations/catalog/crm/microsoft-dynamics-365.md:
```diff
@@ -0,0 +1,218 @@
+
+To correctly send your audience data from Adobe Experience Platform to the [!DNL Microsoft Dynamics 365] destination, you need to go through the field mapping step. Mapping consists of creating a link between your Experience Data Model (XDM) schema fields in your Platform account and their corresponding equivalents from the target destination. To correctly map your XDM fields to the [!DNL Dynamic ...
+
+1. In the "[UICONTROL Mapping]" step, click **[!UICONTROL Add new mapping]**, you will see a new mapping row on the screen.
```
No need for quotes here.

vfalon commented on help/destinations/catalog/crm/microsoft-dynamics-365.md:
```diff
@@ -0,0 +1,218 @@
+1. In the "[UICONTROL Mapping]" step, click **[!UICONTROL Add new mapping]**, you will see a new mapping row on the screen.
+![Add new mapping](../../assets/catalog/crm/microsoft-dynamics-365/add-new-mapping.png)
+
+1. In the "[!UICONTROL Select source field]" window, when selecting the source field choose the **[!UICONTROL Select identity namespace]** category and select `contactId`.
```
No need for quotes.

vfalon commented on help/destinations/catalog/crm/microsoft-dynamics-365.md:
```diff
@@ -0,0 +1,218 @@
+1. Monitor the segment summary and ensure that the count of profiles corresponds to the count created within the segment.
+![Segment](../../assets/catalog/crm/microsoft-dynamics-365/segment.png)
+
+1. Log in to the Dynamics 365 website, then navigate to the **[!DNL Apps]** > **[!DNL Contacts]** page and check if the profiles from the segment have been added. You will also notice the segment status from Experience Platform has been updated against the corresponding custom field attribute that was provided in the **Mapping ID** field during the **[!UICONTROL Activate destination]** > **[!UICO ...
```
Line 187: I think I already mentioned this in my previous comments. There is no [!UICONTROL Activate destination] UI element.

Please rephrase this section as below:

"Log in to the [!DNL Dynamics 365] website, then navigate to the [!DNL Apps] > [!DNL Contacts] page and check if the profiles from the segment have been added. You can see that each segment status in [!DNL Dynamics 365] was updated with the corresponding segment status from Platform, based on the [!UICONTROL Mapping ID] value provided during the segment scheduling step."

mohan-h replied:
We are referring to the Platform [step](https://github.com/AdobeDocs/experience-platform.en/blob/1b3618b96a31bbada9f74e9895b9d546e109c52a/help/destinations/assets/catalog/crm/microsoft-dynamics-365/schedule-segment-export.png). This is to emphasize the corroboration between what was provided in the Platform and what is visible in Dynamics 365 contacts page. 

Have prefixed the word Platform before **[!UICONTROL Activate destination]**.

Any suggestions on what rephrasing is needed ?

vfalon replied:
We only use [!UICONTROL] for actual UI elements that are present in the Experience Platform UI (buttons, fields).
Please see the detailed information in our authoring guide on when to use these tags https://experienceleague.adobe.com/docs/authoring-guide-exl/using/authoring/localization/localize.html?lang=en.

In this section, there is no Activate destination button anywhere in the UI. The segment schedule step is already clearly documented, so you can link to that section.

As far as rephrasing suggestions go, please see my comment above. I have updated it with a clearer explanation.

vfalon commented on help/destinations/catalog/crm/microsoft-dynamics-365.md:
```diff
@@ -0,0 +1,218 @@
+
+| Item | Type | Notes |
+---------|----------|---------|
+| Export type | **[!UICONTROL Profile-based]** | <ul><li>You are exporting all members of a segment, together with the desired schema fields *(for example: email address, phone number, last name)*, according to your field mapping.</li><li> Platform segment statuses are exported to [!DNL Dynamics 365] by specifying their corresponding custom field attribute in [!DNL Dynamics 365], within the **[!U ...
```
On line 69 we need to rephrase the second bullet point:

- Platform segment statuses are exported to [!DNL Dynamics 365] by entering their [!DNL Dynamics 365]  custom field attribute in the [!UICONTROL Mapping ID] field, during the [!UICONTROL Schedule segment export] step.

mohan-h replied:
There is a section #schedule-segment-export-example should I also link to it ?

mohan-h replied:
@vladfalon responded to similar query within https://github.com/AdobeDocs/experience-platform.en/pull/93#discussion_r943427991

vfalon commented on help/destinations/catalog/crm/microsoft-dynamics-365.md:
```diff
@@ -0,0 +1,218 @@
+
+![Error](../../assets/catalog/crm/microsoft-dynamics-365/error.png)
+
+To fix this error, verify that the Mapping ID you provided in [!DNL Dynamics 365] for your Platform segment is valid and exists within [!DNL Dynamics 365].
```
[!UICONTROL Mapping ID]

vfalon commented on help/destinations/catalog/crm/microsoft-dynamics-365.md:
```diff
@@ -0,0 +1,218 @@
+
+When performing the [Schedule segment export](../../ui/activate/activate-segment-streaming-destinations.md) step you must manually map Platform segments to the custom field attribute in [!DNL Dynamics 365].
+
+To do this, select each segment, then enter the corresponding custom field attribute from [!DNL Dynamics 365] in the **[!UICONTROL Mapping ID]** field.
```
No need to bold [!UICONTROL Mapping ID] here, since it's not a procedure. We're only mentioning the field name.

vfalon commented on help/destinations/catalog/crm/microsoft-dynamics-365.md:
```diff
@@ -0,0 +1,218 @@
+
+> [!IMPORTANT]
+>
+> The value used for the Mapping ID should exactly match the name of the custom field attribute created within [!DNL Dynamics 365]. See [[!DNL Dynamics 365] documentation](https://docs.microsoft.com/en-us/dynamics365/customerengagement/on-premises/customize/create-edit-fields?view=op-9-1) if you need guidance on finding your custom field attributes.
```
[!UICONTROL Mapping ID]

vfalon reviewed (changes requested):
Left a few more comments to address.

mohan-h committed:
Updates post review

mohan-h committed:
Updated

mohan-h committed:
Updates

vfalon commented on help/destinations/catalog/crm/microsoft-dynamics-365.md:
```diff
@@ -0,0 +1,218 @@
+
+## Prerequisites {#prerequisites}
+
+### Experience Platform Prerequisites {#prerequisites-in-experience-platform}
```
We use sentence case in headings. "Prerequisites" should be lowercase in this section.

vfalon commented on help/destinations/catalog/crm/microsoft-dynamics-365.md:
```diff
@@ -0,0 +1,218 @@
+
+Refer to Adobe's documentation for [Segment Membership Details schema field group](/help/xdm/field-groups/profile/segmentation.md) if you need guidance on segment statuses.
+
+### Microsoft Dynamics 365 Prerequisites {#prerequisites-destination}
```
Sentence case.

vfalon commented on help/destinations/catalog/crm/microsoft-dynamics-365.md:
```diff
@@ -0,0 +1,218 @@
+
+### Experience Platform Prerequisites {#prerequisites-in-experience-platform}
+
+Before activating data to the [!DNL Dynamics 365] destination, you must have a [schema](/help/xdm/schema/composition.html), a [dataset](https://experienceleague.adobe.com/docs/platform-learn/tutorials/data-ingestion/create-datasets-and-ingest-data.html?lang=en), and [segments](https://experienceleague.adobe.com/docs/platform-learn/tutorials/segments/create-segments.html?lang=en) created in [!DNL  ...
```
Line 25: schema is still linking to an HTML file. Please run a search in the file and replace all HTML URLs with relative Markdown links, where needed.

mohan-h replied:
Odd, distinctly recollect changing it.

vfalon commented on help/destinations/catalog/crm/microsoft-dynamics-365.md:
```diff
@@ -0,0 +1,218 @@
+
+|Target Identity|Example|Description|Considerations|
+|---|---|---|---|
+| `contactId` | 7eb682f1-ca75-e511-80d4-00155d2a68d1 | Unique identifier for a contact.| **Mandatory**, Refer to the [[!DNL Dynamics 365] documentation](https://docs.microsoft.com/en-us/dynamics365/customerengagement/on-premises/developer/entities/contact?view=op-9-1) for further details. |
```
Line 61: Full stop after **Mandatory**.

vfalon commented on help/destinations/catalog/crm/microsoft-dynamics-365.md:
```diff
@@ -0,0 +1,218 @@
+
+To correctly send your audience data from Adobe Experience Platform to the [!DNL Dynamics 365] destination, you need to go through the field mapping step. Mapping consists of creating a link between your Experience Data Model (XDM) schema fields in your Platform account and their corresponding equivalents from the target destination. To correctly map your XDM fields to the [!DNL Dynamics 365] des ...
+
+1. In the [UICONTROL Mapping] step, click **[!UICONTROL Add new mapping]**, you will see a new mapping row on the screen.
```
Line 126: Full stop instead of comma after "click **[!UICONTROL Add new mapping]**,".

vfalon commented on help/destinations/catalog/crm/microsoft-dynamics-365.md:
```diff
@@ -0,0 +1,218 @@
+1. In the [UICONTROL Mapping] step, click **[!UICONTROL Add new mapping]**, you will see a new mapping row on the screen.
+![Platform UI screenshot example for Add new mapping.](../../assets/catalog/crm/microsoft-dynamics-365/add-new-mapping.png)
+
+1. In the [!UICONTROL Select source field] window, when selecting the source field choose the **[!UICONTROL Select identity namespace]** category and select `contactId`.
```
Line 126: Please remove the "when selecting the source field" part.

vfalon commented on help/destinations/catalog/crm/microsoft-dynamics-365.md:
```diff
@@ -0,0 +1,218 @@
+1. Monitor the segment summary and ensure that the count of profiles corresponds to the count created within the segment.
+![Platform UI screenshot example showing Segment.](../../assets/catalog/crm/microsoft-dynamics-365/segment.png)
+
+1. Log in to the Dynamics 365 website, then navigate to the **[!DNL Apps]** > **[!DNL Contacts]** page and check if the profiles from the segment have been added. You will also notice the segment status from Experience Platform has been updated against the corresponding custom field attribute that was provided in the **Mapping ID** field during the Platform **[!UICONTROL Activate destination]** > ...
```
Line 187: I think I already mentioned this in my previous comments. There is no **[!UICONTROL Activate destination]** UI element. 

Please rephrase this section as below:

"Log in to the [!DNL Dynamics 365] website, then navigate to the [!DNL Customers] > [!DNL Contacts] page and check if the profiles from the segment have been added. You can see that each segment status in [!DNL Dynamics 365] was updated with the corresponding segment status from Platform, based on the [!UICONTROL Mapping ID] value provided during the segment scheduling step."

mohan-h replied:
Had responded with the substantiation and requested for your guidance https://github.com/AdobeDocs/experience-platform.en/pull/90#discussion_r942611574 replicated below.

We are referring to the Platform [step](https://github.com/AdobeDocs/experience-platform.en/blob/1b3618b96a31bbada9f74e9895b9d546e109c52a/help/destinations/assets/catalog/crm/microsoft-dynamics-365/schedule-segment-export.png). This is to emphasize the corroboration between what was provided in the Platform and what is visible in Dynamics 365 contacts page. 

Have prefixed the word Platform before **[!UICONTROL Activate destination]**.

Any suggestions on what rephrasing is needed ?

vfalon replied:
We only use [!UICONTROL] for actual UI elements that are present in the Experience Platform UI (buttons, fields).
Please see the detailed information in our authoring guide on when to use these tags https://experienceleague.adobe.com/docs/authoring-guide-exl/using/authoring/localization/localize.html?lang=en.

In this section, there is no _Activate destination_ button anywhere in the UI. The segment schedule step is already clearly documented, so you can link to that section.

As far as rephrasing suggestions go, please see my comment above. I have updated it with a clearer explanation.

vfalon commented on help/destinations/catalog/crm/microsoft-dynamics-365.md:
```diff
@@ -0,0 +1,218 @@
+* [IOrganizationService.Update(Entity) Method](https://docs.microsoft.com/en-us/dotnet/api/microsoft.xrm.sdk.iorganizationservice.update?view=dataverse-sdk-latest)
+* [Update and delete table rows using the Web API](https://docs.microsoft.com/en-us/power-apps/developer/data-platform/webapi/update-delete-entities-using-web-api#basic-update)
+
+### Limits
```
Please move the limits section after the Prerequisites, and call it Guardrails.

vfalon reviewed (changes requested):
Please see my comments, as it seems some previous ones were missed.

mohan-h committed:
Merge branch 'AdobeDocs:main' into msdynamics-dest

vfalon commented on help/destinations/catalog/crm/microsoft-dynamics-365.md:
```diff
@@ -0,0 +1,218 @@
+1. Monitor the segment summary and ensure that the count of profiles corresponds to the count created within the segment.
+![Platform UI screenshot example showing Segment.](../../assets/catalog/crm/microsoft-dynamics-365/segment.png)
+
+1. Log in to the Dynamics 365 website, then navigate to the **[!DNL Apps]** > **[!DNL Contacts]** page and check if the profiles from the segment have been added. You will also notice the segment status from Experience Platform has been updated against the corresponding custom field attribute that was provided in the **Mapping ID** field during the Platform **[!UICONTROL Activate destination]** > ...
```
Line 187: Based on the screenshot, the UI path is Customers -> Contacts, not **[!DNL Apps]** > **[!DNL Contacts]**. 

Please make sure to follow the UI of the destination platforms when copying content between destination pages.

mohan-h replied:
Thanks, You are correct but the last time the navigation the engineer and testing teams had shown was navigating to the home page at the left corner i.e. the Apps icons and then searching to find Contacts. Maybe I am recollecting incorrectly.

vfalon commented on help/destinations/catalog/crm/microsoft-dynamics-365.md:
```diff
@@ -0,0 +1,218 @@
+
+        > [!IMPORTANT]
+        >
+        > When mapping to a [!DNL Dynamics 365] [date or timestamp](https://docs.microsoft.com/en-us/power-apps/developer/data-platform/webapi/reference/timestampdatemapping?view=dataverse-latest) attribute, the corresponding attribute value in the dataset should not be empty. During execution if the value is empty you will encounter errors.
```
Line 146: "the corresponding attribute value in the dataset should not be empty."

Which dataset? Whici attribute value? The one in Dynamics or the one in Platform? Please clarify this note with engineering, as it sounds confusing in this state. Reading it now, I don't really understand what it refers to.

mohan-h replied:
Have rephrased.

mohan-h committed:
Updated post review

mohan-h committed:
Merge branch 'AdobeDocs:main' into msdynamics-dest

mohan-h committed:
Content updates

vfalon commented on help/destinations/catalog/crm/microsoft-dynamics-365.md:
```diff
@@ -0,0 +1,218 @@
+
+        > [!IMPORTANT]
+        >
+        > If you have a date or timestamp source field which is mapped to a [!DNL Dynamics 365] [date or timestamp](https://docs.microsoft.com/en-us/power-apps/developer/data-platform/webapi/reference/timestampdatemapping?view=dataverse-latest) target field, ensure that the value being passed should not be empty. If the value passed is empty you will encounter a *`Bad request reported while pushi ...
```
Line 150: "[...] ensure that the mapped value is not empty. [...] This is a [!DNL Dynamics 365] limitation."

vfalon reviewed (changes requested):
The PR looks good! Just one comment to address.

mohan-h committed:
Updates post review

Updates against https://github.com/AdobeDocs/experience-platform.en/pull/90#discussion_r966910628

mohan-h committed:
Merge branch 'AdobeDocs:main' into msdynamics-dest

mohan-h committed:
Changes for new authentication approach

mohan-h committed:
Multiple Environment URL input no longer needed

mohan-h committed:
Minor formatting changes

mohan-h committed:
missing formatting around UICONTROL added

mohan-h committed:
Changed click to select

mohan-h committed:
Added missing !DNL and !UICONTROL directives

mohan-h committed:
Formatting update

vfalon commented on help/destinations/catalog/crm/microsoft-dynamics-365.md:
```diff
@@ -0,0 +1,218 @@
+
+This [!DNL Adobe Experience Platform] [destination](/help/destinations/home.md) leverages the [[!DNL Contact Entity Reference API]](https://docs.microsoft.com/en-us/dynamics365/customerengagement/on-premises/developer/entities/contact?view=op-9-1), which allows you to update identities within a segment into [!DNL Dynamics 365].
+
+[!DNL Dynamics 365] uses OAuth 2 with Authorization Grant as the authentication mechanism to communicate with the [!DNL Contact Entity Reference API]. Instructions to authenticate to your [!DNL Dynamics 365] instance are further below, within [Authenticate to destination](#authenticate) section.
```
"[...]in the [Authenticate to destination](#authenticate) section."

mohan-h replied:
For info from VS Code : `After 'the', the verb 'Authenticate' doesn't fit. Is 'Authenticate' spelled correctly? If 'Authenticate' is the first word in a compound adjective, use a hyphen between the two words. Using the verb 'Authenticate' as a noun may be non-standard.LTeX[A_INFINITIVE](https://community.languagetool.org/rule/show/A_INFINITIVE?lang%3Den-US)`

vfalon replied:
Ignore that VS code info, please. Use the version I suggested. It's correct, since it's the name of the section. VS Code doesn't understand those nuances.

vfalon commented on help/destinations/catalog/crm/microsoft-dynamics-365.md:
```diff
@@ -0,0 +1,218 @@
+
+#### Register an application and application user within Azure Active Directory {#prerequisites-app-user}
+
+To enable [!DNL Dynamics 365] to access resources you will need to log in with your [!DNL Azure Account] to [[!DNL Azure Active Directory]](https://docs.microsoft.com/en-us/azure/active-directory/develop/howto-create-service-principal-portal#register-an-application-with-azure-ad-and-create-a-service-principal) and create the below:
```
[...] and create the following:

- An Azure Active Directory application
- A service principal
- An application secret

mohan-h replied:
Also updated all occurrences of Azure AD > Azure Active Directory

vfalon commented on help/destinations/catalog/crm/microsoft-dynamics-365.md:
```diff
@@ -0,0 +1,218 @@
+To authenticate to the destination, select **[!UICONTROL Connect to destination]**.
+![Platform UI screenshot showing how to authenticate.](../../assets/catalog/crm/microsoft-dynamics-365/authenticate-destination.png)
+
+Fill in the required fields below. Refer to the [Gather Dynamics 365 credentials](#gather-credentials) section for any guidance.
```
Let's use the order of fields as shown in the screenshot.

vfalon commented on help/destinations/catalog/crm/microsoft-dynamics-365.md:
```diff
@@ -0,0 +1,218 @@
+| `Tenant ID` | The [!DNL Dynamics 365] Tenant ID for your [!DNL Azure AD] application. Refer to the [[!DNL Dynamics 365] documentation](https://docs.microsoft.com/en-us/azure/active-directory/develop/howto-create-service-principal-portal#get-tenant-and-app-id-values-for-signing-in) for guidance.  | `1234567-aaaa-12ab-ba21-1234567890` |
+| `Environment URL` | Refer to the [[!DNL Dynamics 365] documentation](https://docs.microsoft.com/en-us/dynamics365/customerengagement/on-premises/developer/org-service/discover-url-organization-organization-service?view=op-9-1) for guidance.| If your [!DNL Dynamics 365] domain is as below, you need the highlighted value.<br> *`org57771b33`.crm.dynamics.com* |
+
+## Guardrails
```
Add an anchor tag to Guardrails, please.

mohan-h replied:
Oops, great spot.

vfalon reviewed (changes requested):
Looks good! Left a few minor comments.

mohan-h committed:
Review updates

mohan-h committed:
Merge branch 'AdobeDocs:main' into msdynamics-dest

vgiurgiu commented:
Closing, since this documentation was already published. Many thanks for your work, @mohan-h !
